File: app/src/main/java/com/mmjang/ankihelper/data/quote/QuoteDb.java
package com.mmjang.ankihelper.data.quote;

import android.content.Context;
import android.database.Cursor;

import java.util.Random;

/**
 * Created by liao on 2017/8/13.
 */

public class QuoteDb {
    private static final String DATABASE_NAME = "quote.db";
    private static final int DATABASE_VERSIOn = 1;
    private static final int ID_MAX = 15972;

    private static QuoteDb instance = null;
    private QuoteDatabaseHelper db; // Holds the database helper instance
    private Context mContext;

    protected QuoteDb(Context context){
        // Initialize the database helper
        db = new QuoteDatabaseHelper(context);
        mContext = context;
    }

    public static QuoteDb getInstance(Context context){
        if(instance == null){
            instance = new QuoteDb(context);
        }
        return instance;
    }

    public String getQuote() {
        int randomKey = randInt(0, ID_MAX);
        // Get the quote from the database
        Cursor cursor = db.getReadableDatabase().query("quote", new String[]{"content"}, "id=?", new String[]{Integer.toString(randomKey)}, null, null, null);
        String content = "";
        if (cursor.moveToNext()) {
            content = cursor.getString(0);
        }
        return content;
    }

    public static int randInt(int min, int max) {

        Random rand = new Random();
        int randomNum = rand.nextInt((max - min) + 1) + min;
        return randomNum;
    }
}

File: app/src/main/java/com/lmyby/ankihelper/data/quote/QuoteDb.kt
package com.lmyby.ankihelper.data.quote

import android.content.Context
import kotlin.random.Random

/**
 * Created by liao on 2017/8/13.
 * Converted to Kotlin as part of Phase 2 data model migration
 */
class QuoteDb private constructor(context: Context) {
    private val db: QuoteDatabaseHelper = QuoteDatabaseHelper(context)
    private val mContext: Context = context

    fun getQuote(): String {
        val randomKey = randInt(0, ID_MAX)
        val database = db.readableDatabase
        val cursor = database.query(
            "quote",
            arrayOf("content"),
            "id=?",
            arrayOf(randomKey.toString()),
            null,
            null,
            null
        )
        var content = ""
        if (cursor.moveToNext()) {
            content = cursor.getString(0)
        }
        cursor.close()
        return content
    }

    companion object {
        private const val DATABASE_NAME = "quote.db"
        private const val DATABASE_VERSION = 1
        private const val ID_MAX = 15972

        @Volatile
        private var instance: QuoteDb? = null

        @JvmStatic
        fun getInstance(context: Context): QuoteDb {
            return instance ?: synchronized(this) {
                instance ?: QuoteDb(context).also { instance = it }
            }
        }

        @JvmStatic
        fun randInt(min: Int, max: Int): Int {
            return Random.nextInt(min, max + 1)
        }
    }
}
